controllers/chat/group/groupAddMembers.js

```diff
@@ -61,29 +61,34 @@ const groupAddMembers = async (req, res) => {
   try {
     const responseChannel = await channelToAdd?.addMembers(filteredMembersObject);
 
-    filteredMembersId.map(async (member) => {
-      const targetUserModel = await UsersFunction.findUserById(User, member);
-
-      const textOwnUser = `You added ${targetUserModel.username} to this group`;
-      const textTargetUser = `${ownUser.username} added you to this group`;
-      const textDefaultUser = `${ownUser.username} added ${targetUserModel.username} to this group`;
-
-      await channelToAdd.sendMessage({
-        text: textDefaultUser,
-        own_text: textOwnUser,
-        other_text: textTargetUser,
-        other_system_user: member,
-        better_type: 'add_member_to_group',
-        type: 'system',
-        user_id: userId,
-        only_to_user_show: userId,
-        disable_to_user: false,
-        is_from_prepopulated: true,
-        system_user: userId,
-        isSystem: true,
-        members: [member]
-      });
-    });
+    filteredMembersId.map(
+      async (member) => {
+        const targetUserModel = await UsersFunction.findUserById(User, member);
+
+        const textOwnUser = `You added ${targetUserModel.username} to this group`;
+        const textTargetUser = `${ownUser.username} added you to this group`;
+        const textDefaultUser = `${ownUser.username} added ${targetUserModel.username} to this group`;
+
+        await channelToAdd.sendMessage({
+          text: textDefaultUser,
+          own_text: textOwnUser,
+          other_text: textTargetUser,
+          other_system_user: member,
+          better_type: 'add_member_to_group',
+          type: 'system',
+          user_id: userId,
+          only_to_user_show: userId,
+          disable_to_user: false,
+          is_from_prepopulated: true,
+          system_user: userId,
+          isSystem: true,
+          members: [member]
+        });
+      },
+      {
+        skip_push: true
+      }
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `Array.prototype.map` function is used incorrectly here. The `map` function is intended to create a new array with the results of calling a provided function on every element in the calling array. However, in this case, you're using it for its side effects (sending messages), not for transforming an array. This can lead to unexpected behavior, especially when dealing with asynchronous operations.

Instead, consider using `Array.prototype.forEach` if you don't care about the order of execution, or `for...of` loop along with `async/await` if you want to send messages sequentially.

Also, the `{ skip_push: true }` object seems to be misplaced. It should be part of the `sendMessage` method call as an option.

Here's how you could refactor your code:

```diff
-    filteredMembersId.map(
+    for (const member of filteredMembersId) {
-      async (member) => {
         const targetUserModel = await UsersFunction.findUserById(User, member);

         const textOwnUser = `You added ${targetUserModel.username} to this group`;
         const textTargetUser = `${ownUser.username} added you to this group`;
         const textDefaultUser = `${ownUser.username} added ${targetUserModel.username} to this group`;

-        await channelToAdd.sendMessage({
+        await channelToAdd.sendMessage({
           text: textDefaultUser,
           own_text: textOwnUser,
           other_text: textTargetUser,
           other_system_user: member,
           better_type: 'add_member_to_group',
           type: 'system',
           user_id: userId,
           only_to_user_show: userId,
           disable_to_user: false,
           is_from_prepopulated: true,
           system_user: userId,
           isSystem: true,
           members: [member]
-        });
-      },
-      {
-        skip_push: true
-      }
+        }, { skip_push: true });
     }
```
This change will ensure that the messages are sent sequentially and the `skip_push` option is correctly passed to the `sendMessage` method.


<!-- This is an auto-generated comment by OSS CodeRabbit -->